scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -128,6 +130,9 @@
 	/** does the server require the client to authenticate */
 	private Boolean clientAuthenticationRequired;
 
+	/** does not start handshakes */
+	private Boolean serverOnly;
```

Comment:
Should we rather go for an enum to anticipate clientOnly ? (Client, Server, Both)

(This make me think again to https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/410 and sry to insist but I really like to know your opinion about that :pray:)

Reply:
I'm simply not up-to-date with LWM2M,  especially 1.1.

Reply:
> the client only flag is only used by the builder to verify the configuration. the server only limits the connector function.

I didn't notice that we already have a client only flag. I just think that this could make sense to have the same restriction for client side (I mean limiting connector function). and maybe we could anticipate it.

> I'm simply not up-to-date with LWM2M, especially 1.1.

Just talking about LWM2M 1.0, I know that by the past you didn't like this idea of DTLS exchange role. And I would like to know, if you see a way to handle the failover use case for none queue mode ?

Reply:
> by the past you didn't like this idea of DTLS exchange role

Yes, but that is mainly based on the fact, that I don't consider to have stable ip-adresses :-).

Reply:
Ok, so for stable ip-address, DTLS exchanges roles seems not so bad to you ? To be honest I can not see other way for now ... (except maybe DTLS heartbeat which means more bandwidth consumption ...)

Reply:
If a DTLS role exchange works, OK. But in my experience, the CLIENT_HELLO from the server usually mixup the people (and devices) more, than it works :-). Therefore I would prefer to just add the possibility to switch it off.

Reply:
I totally agree that make is switchable is a good idea.

But the initial remarks was : 

1. Does it makes sense to also have a client only mode ?" (I mean exactly like you implement the serverOnly)
2. If yes, maybe we should anticipate it and use an enum instead of a boolean ? or should we reuse the actual ClientOnly to also limit connector behavior ? (not necessarily in this PR)

Reply:
AFAIK the embedded device are built without the dtls-server-code to save code memory. But for californium, that is not really worth :-).
So my intention was really just to ensure, that the server doesn't start a handshake. I want to enusre this also for the unit test, which may fail, if the server accidently and unexpected start a handshake for which ever reason.

Reply:
Ok let's this aside for now, but I still feel that this could make sense to set a DTLS connector in a way it refuse (CLIENT_HELLO) and so act as a client only (At least for testing purpose)